### PR TITLE
[Patient Intake Sample] Bots setup 

### PR DESCRIPTION
**Pull request by filipeximenes**
# Description
- Adds page to upload bots
- Adds `build:bots` command
- Fixes handler function parameter order
- Improves handling of `undefined` form fields

### CODE REVIEW

File: examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx [hunk 1]
```diff
@@ -72,10 +66,130 @@ export function UploadDataPage(): JSX.Element {
...
-  const result = await medplum.executeBatch(data);
+async function uploadCoreData(medplum: MedplumClient): Promise<void> {
+  const batch = coreData as Bundle;
+
+  const result = await medplum.executeBatch(batch);
+
+  showNotification({
+    icon: <IconCircleCheck />,
+    title: 'Success',
+    message: 'Uploaded Business Statuses',
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx:77]: "Business Status" is not the right description here. We need to update this message

File: examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx [hunk 2]
```diff
@@ -72,10 +66,130 @@ export function UploadDataPage(): JSX.Element {
...
+  });
+
+  if (result.entry?.every((entry) => entry.response?.outcome && isOk(entry.response?.outcome))) {
+    await setTimeout(
+      () =>
+        showNotification({
+          icon: <IconCircleCheck />,
+          title: 'Success',
+          message: 'Uploaded Business Statuses',
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx:86]: "Business Status" is not the right description here. We need to update this message

File: examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx (hunk 1 above)
Comment (rahul1) [reply examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx:77]: Why do we need this notificaiton, as we have another success notification down below

File: examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx [hunk 3]
```diff
@@ -72,10 +66,130 @@ export function UploadDataPage(): JSX.Element {
...
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx:111]:
```suggestion
          message: 'Uploaded Example Data',
```

File: examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx [hunk 4]
```diff
@@ -72,10 +66,130 @@ export function UploadDataPage(): JSX.Element {
...
+    title: 'Success',
+    message: 'Deployed Example Bots',
+  });
+}
+
+function checkQuestionnairesUploaded(medplum: MedplumClient): boolean {
+  let check = false;
+  const intakeQuestionnaire = medplum.searchResources('Questionnaire').read();
+
+  const questionnairesToCheck = intakeQuestionnaire.filter((questionnaire) => questionnaire.name === 'patient-intake');
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx:177]: You should be able to incorporate the name search into line 175

File: examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts [hunk 5]
```diff
@@ -16,47 +16,40 @@ import {
...
-    given: [answers['first-name'].valueString, answers['middle-name'].valueString],
-    family: answers['last-name'].valueString,
-  } as HumanName;
-
-  patient.name = [newName];
-  patient.birthDate = answers['dob'].valueDate;
-  patient.gender = answers['gender-identity'].valueCoding?.code as Patient['gender'];
+  patient.name = [
+    {
+      given: [answers['first-name']?.valueString, answers['middle-name']?.valueString],
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts:38]:
```
 answers['middle-name']?.valueString],
```

This potentially adds `undefined` to the array. we should filter out this undefined

File: examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts [hunk 6]
```diff
@@ -105,6 +98,9 @@ export async function handler(event: BotEvent<QuestionnaireResponse>, medplum: M
     { valueDateTime: convertDateToDateTime(answers['estimated-delivery-date'].valueDate) }
   );
 
+  // Handle coverage
+
+  const questionnaire = await medplum.readResource('Questionnaire', (response.questionnaire as string).split('/')[1]);
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts:103]: We have helper function

```suggestion
  const questionnaire = await medplum.readResource('Questionnaire', resolveId(response.questionnaire));
```

File: examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx [hunk 7]
```diff
@@ -72,10 +64,114 @@ export function UploadDataPage(): JSX.Element {
...
+        }),
+      1000
+    );
+  } else {
+    throw new Error('Error uploading example data');
+  }
+}
+
+async function uploadExampleBots(medplum: MedplumClient, profile: Practitioner): Promise<void> {
+  const questionnaires = await medplum.searchResources('Questionnaire');
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx:107]: @filipeximenes we should be able to use the questionnaire context we set up in App.tsx

File: examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx [hunk 8]
```diff
@@ -1,66 +1,58 @@
...
-import { useParams } from 'react-router-dom';
+import { useNavigate, useParams } from 'react-router-dom';
 import coreData from '../../data/core/patient-intake-questionnaire.json';
 import exampleData from '../../data/example/example-patient-data.json';
+import exampleBotData from '../../data/core/example-bots.json';
 
-type UploadFunction = (medplum: MedplumClient, data: Bundle) => Promise<void>;
+type UploadFunction =
+  | ((medplum: MedplumClient, profile: Practitioner) => Promise<void>)
+  | ((medplum: MedplumClient) => Promise<void>);
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/pages/UploadDataPage.tsx:15]:
```suggestion
  ((medplum: MedplumClient, profile?: Practitioner) => Promise<void>);
```

**Review by rahul1 — APPROVED** [review]

**Comment by rahul1** [comment]
A few comments, but then feel free to merge

File: examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts (hunk 5 above)
Comment (filipeximenes) [reply examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts]: Move this logic to a function to make name processing more fine grained

File: examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts (hunk 6 above)
Comment (rahul1) [reply examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts]: @filipeximenes you're right, my mistake. There's a bit of an edge case for why `response.questionnaire`.

Actually, I think the preferred method would be 
```ts
const questionnaire = await medplum.readReference({ reference: response.questionnaire });
```

**Review by rahul1 — COMMENTED** [review]
One minor comment, then looks good to me

**Pull request merged**